### PR TITLE
React16

### LINKED ISSUE #498: React 16 compatibility
### Summary:

I looked around in the issue tracker but couldn't find an existing issue. As React 16 is right around the corner, maybe it's time to prepare a PR that adds support for it?

It looks like at least the [call to `unstable_renderSubtreeIntoContainer`](https://github.com/reactjs/react-modal/blob/f007aeb7336bc8a47bb6b8c7919ed3a077fbd87d/src/components/Modal.js#L11) should be updated as there's now first class support for portals.

### Steps to reproduce:

 1. Render a modal.
 2. Unmount it.

### Expected behavior:

Shouldn't throw any errors.

### Link to example of issue:

https://codesandbox.io/s/xpl8r853o4

### PR DESCRIPTION
**Pull request by catamphetamine**
Fixes #498

Based on the implementation on [CogniStreamer/react-modal@4434c4a](https://github.com/CogniStreamer/react-modal/commit/4434c4aac57d91a24d8f1b89e85a41fe2bd2023d) 
See all details there.

### CODE REVIEW

File: src/components/Modal.js [hunk 1]
```diff
@@ -8,6 +8,8 @@ import SafeHTMLElement from '../helpers/safeHTMLElement';
 export const portalClassName = 'ReactModalPortal';
 export const bodyOpenClassName = 'ReactModal__Body--open';
 
+const useReactPortal = ReactDOM.createPortal !== undefined;
```
Comment (nickhudkins) [inline src/components/Modal.js]: Should we introduce a warning here indicating that usage without ReactDOM.createPortal is now deprecated and will be removed in the next major release?

Comment (catamphetamine) [reply src/components/Modal.js]: Then every React 15 user will get those warnings so they'll create issues about disabling those warnings.
Most likely some folks out there are using legacy React 14 even.

Comment (diasbruno) [reply src/components/Modal.js]: @catamphetamine @nickhudkins Remember that all the changes to support react 16 must target the branch `next`. This way we can safely make any changes.

Comment (marlonbernardes) [reply src/components/Modal.js]: What about upgrading to `3.0.0` and supporting only React 16+?

Comment (diasbruno) [reply src/components/Modal.js]: @marlonbernardes It would be preferable to keep it backwards compatible. Also, to avoid maintaining 2 versions.

**Comment by nickhudkins** [comment]
3.0.0 seems like a good opportunity to do some heavier work (if there is
work to be done) but I think backwards compatible and encouragement for
people to keep moving forward is a good path

File: src/components/Modal.js (hunk 1 above)
Comment (marlonbernardes) [reply src/components/Modal.js]: It makes perfect sense. Let me know if there's anything I can help you with.

**Comment by catamphetamine** [comment]
I added @diasbruno to collaborators to the repo:
https://github.com/catamphetamine/react-modal
Anyway this PR is gonna be closed I guess because I heard that there will be a special `next` branch for these changes.
If anyone wants to take part committing code then leave a comment here requesting to be added to collaborators.

**Comment by diasbruno** [comment]
@catamphetamine Please, can you change the target branch to be `next`?
 
@nickhudkins Agreed. Once we are done with `next` and we are ready to merge with `master`,  the new release will be `3.0`.

**Comment by catamphetamine** [comment]
@diasbruno Ok, I'm closing this PR and creating a new one then

**Comment by catamphetamine** [comment]
Seems that closing PR is not required to switch branches.
That's good feature.
Switched to `next`.

**Comment by tomasztomys** [comment]
Can you release new version with this PR asap?

**Comment by catamphetamine** [comment]
@tomasztomys This PR was not accepted by the maintainers of the library so I should actually close not to confuse other users.
Consult the developers of this library on the estimated progress on this issue.
I'm no longer involved.

**Comment by lawlmart** [comment]
Sorry if I'm confused, but shouldn't this have been merged into next instead of closed?

**Comment by catamphetamine** [comment]
@lawlmart The developers of this library decided to take some other approach.
https://github.com/catamphetamine/react-modal/pull/1#issuecomment-333341305
Something related to refactoring.

**Pull request closed without merging**